ctapipe/reco/shower_reconstructor.py

```diff
@@ -0,0 +1,208 @@
+from functools import partial
+
+import astropy.units as u
+import numpy as np
+from astropy.coordinates import AltAz, SkyCoord
+from iminuit import Minuit
+
+from ctapipe.containers import Model3DReconstructedGeometryContainer
+from ctapipe.coordinates import CameraFrame, TelescopeFrame
+from ctapipe.core.traits import Unicode
+from ctapipe.image import (
+    GaussianShowermodel,
+    ShowermodelPredictor,
+    neg_log_likelihood_approx,
+)
+from ctapipe.reco.reco_algorithms import Reconstructor
+
+
+class Model3DGeometryReconstructor(Reconstructor):
+    geometry_seed = Unicode(default_value="HillasReconstructor").tag(config=True)
+
+    def __init__(self, subarray, **kwargs):
+        super().__init__(subarray=subarray, **kwargs)
+
+    def __call__(self, event):
+        if self.geometry_seed not in event.dl2.stereo.geometry:
+            raise ValueError()
+
+        # get all telescope properties that we need for the ShowermodelPredictor
+        tel_positions = {}
+        tel_solid_angles = {}
+        tel_mirror_area = {}
+        tel_spe_widths = {}
+        tel_pedestial_widths = {}
+        for tel_id in event.dl1.tel.keys():
+            tel_positions[tel_id] = self.subarray.positions[tel_id].to_value(u.m)
+            geometry = self.subarray.tel[tel_id].camera.geometry
+            tel_solid_angles[tel_id] = geometry.transform_to(
+                TelescopeFrame()
+            ).pix_area.to_value(u.rad**2)
+            tel_mirror_area[tel_id] = self.subarray.tel[
+                tel_id
+            ].optics.mirror_area.to_value(u.m**2)
+            tel_spe_widths[tel_id] = (
+                spe
+                if (spe := event.mon.tel[tel_id].flatfield.charge_std) is not None
+                else 0.5
```

Comment:
magic number

Reply:
So the problem is that the simulations do not contain the flatfield and pedestal charge information. I need some values in case they are not provided and I don't have a clue which numbers are suitable. I think this would also vary with telescope type. It's also part of the impact code https://github.com/cta-observatory/ctapipe/blob/76c672ed2f8e66b0a31bae8bd53350c1a81c55a5/ctapipe/reco/impact.py#L120